sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -711,6 +712,23 @@ object DataSource extends Logging {
     }
   }
 
+  /**
+   * Return a [[TableProvider]] instance for the given provider. It returns None if the provider
```

Comment:
`Return a [[TableProvider]] instance for the given provider. It returns None if the provider is not Data Source V2, or it's configured to fallback to Data Source V1 code path.`
=>
`Returns an optional [[TableProvider]] instance for the given provider. It returns None if there is no corresponding Data Source V2 implementation, or the provider is configured to fallback to Data Source V1 code path.`